### DISCUSSION TITLE
What kind of tutorials would you like to see?

### DISCUSSION THREAD

**Discussion by zerebos** [TriliumNext/Trilium#4259] [Q&A]
I'm working on improving some of the documentation and wiki for Trilium including adding some tutorials for addon development. I already have a couple written that I need to add that focus on making custom widgets. I have a handful of other ideas of what tutorial topics might be helpful, but I'd like to hear thoughts from the community.

So please let me know any and all ideas you might have for tutorials, or even things you'd like to see improved or expanded on for the documentation.

**Comment by sottey** [comment]
Hi @rauenzi ,

Thank you for the offer! I have begun keeping snippets in Trilium for different things, but would love to see more tutorials on the API, Scoping CSS (avoiding impacting other stuff in Trilium with CSS in rendered notes or widgets).

If you are looking for help, I would be happy to take a topic you recommend and take a swing at writing a tutorial for you as well.

Thanks!
Sean

**Reply by zerebos** [reply]
Based on your other comment, do you mean ETAPI when you say API? Or are you including the frontend and backend APIs?

**Comment by sottey** [comment]
Oh, one more thing regarding documentation....

I may have missed it, but the only API docs I see are the openAPI YAML. I had to figure out how to convert to workable html (python, openapi standalone script, etc.). I now have a static html doc with all the information, but we might want to host that in the docs for easier reference.

**Reply by zerebos** [reply]
You're right that the YAML file is the only ETAPI documentation currently, that's definitely something I could expand on in the wiki. But both the frontend and backend API have their own documentation pages already.

**Reply by sottey** [reply]
Totally agree. I have used the wiki entries quite a bit. But I love [this](https://elzs.net/triliumapi.html)

**Reply by zerebos** [reply]
Yeah I also used swagger to help visualize the ETAPI, that's how I was able to build [trilium-etapi](https://github.com/rauenzi/trilium-etapi)! Could definitely add something like that for the ETAPI.

**Reply by zerebos** [reply]
I've put a public instance of `swagger-ui` for the ETAPI here: https://trilium.rocks/etapi (alias subject to change, link with noteId: https://trilium.rocks/pPIXi0uwF5GX)

**Comment by zerebos** [comment]
As an update to this, I have added two new tutorials to the wiki. [Frontend Basics](https://github.com/zadam/trilium/wiki/Frontend-Basics) and [Widget Basics](https://github.com/zadam/trilium/wiki/Widget-Basics)
